components/arm/xarm/xarm.go

```diff
@@ -168,9 +172,7 @@ func (x *xArm) Reconfigure(ctx context.Context, deps resource.Dependencies, conf
 		}
 	}
 
-	if newConf.Speed > 0 {
-		x.speed = float32(utils.DegToRad(float64(speed)))
-	}
+	x.speed = float32(utils.DegToRad(float64(speed)))
```

Comment:
I think if you're going to do this you should add a `<=` check on L148

Reply:
I think the better thing to do there is error on negative speed, not overwrite it.

Reply:
Yeah this is better I think